### PR TITLE
Require JSX without extension

### PR DESCRIPTION
**Pull request by islemaster**
Updates our build configuration to allow us to require JSX files without including the extension.

``` javascript
// Before:
var Component = require('./Component.jsx');
// After:
var Component = require('./Component');
```

To get this working everywhere I enabled it for browserify-inc in apps, mochify in apps, and browserify in code-studio.

I'm making this change this due to a suspicion that the issue in #7737 might be related to this (though I'm starting to doubt it) and because we've talked about doing it for a while.

### CODE REVIEW

File: apps/Gruntfile.js [hunk 1]
```diff
@@ -275,6 +275,7 @@ module.exports = function (grunt) {
       ' `npm bin`/browserifyinc' +
       ' -g [ browserify-global-shim ]' +
       ' --cachefile ' + outputDir + 'browserifyinc-cache.json' +
+      ' --extension=.jsx' +
```
Comment (islemaster) [inline apps/Gruntfile.js:278]: Usage [as documented here](https://github.com/substack/node-browserify#usage).

File: code-studio/scripts/build-commands.js [hunk 2]
```diff
@@ -80,6 +81,9 @@ exports.bundle = function (config) {
     // Enables source map
     debug: true,
 
+    // Allow requiring jsx without specifying extension
+    extensions: ['.jsx'],
```
Comment (islemaster) [inline code-studio/scripts/build-commands.js:85]: When using the API, the argument is `extensions` (plural) [as documented here](https://github.com/substack/node-browserify#methods).

File: apps/test/util/runTests.js [hunk 3]
```diff
@@ -70,6 +70,7 @@ exec(command, function (err, stdout, stderr) {
   mochify(globs.join(' '), {
     grep: grep,
     debug: process.env.mocha_debug,
+    extension: ['.jsx'],
```
Comment (islemaster) [inline apps/test/util/runTests.js:73]: However, the mochify API option is `extension` (singular) to match the long-form CLI option, [as more-or-less documented here](https://github.com/mantoni/mochify.js#api).

**Comment by Bjvanminnen** [comment]
Nice! I've tried and failed to make this change before :) lgtm

**Pull request merged**